### PR TITLE
Separate perf tests into a different travis check

### PR DESCRIPTION
**Pull request by vtomole**
Fixes: https://github.com/quantumlib/Cirq/issues/1597

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -46,4 +46,13 @@ matrix:
     install:
       - pip install -r requirements.txt
       - pip install -r dev_tools/conf/pip-list-dev-tools.txt
-    script: check/pytest --ignore=cirq/contrib
+    script: check/pytest --ignore=cirq/contrib --benchmark-skip
+
+
```
Comment (viathor) [inline .travis.yml]: nit: these other blocks are separated by a single newline, so let's remove one here for consistency.

File: .travis.yml [hunk 2]
```diff
@@ -46,4 +46,13 @@ matrix:
...
+    - os: linux
+    env: NAME=Performance tests
+    python: "3.7"
+    install:
+      - pip install -r requirements.txt
+      - pip install -r dev_tools/conf/pip-list-dev-tools.txt
+    script: check/pytest --benchmark-only
```
Comment (viathor) [inline .travis.yml]: We should probably keep `--ignore=cirq/contrib` here.

File: .travis.yml [hunk 3]
```diff
@@ -46,4 +46,13 @@ matrix:
     install:
       - pip install -r requirements.txt
       - pip install -r dev_tools/conf/pip-list-dev-tools.txt
-    script: check/pytest --ignore=cirq/contrib
+    script: check/pytest --ignore=cirq/contrib --benchmark-skip
+
+
+    - os: linux
+    env: NAME=Performance tests
+    python: "3.7"
```
Comment (viathor) [inline .travis.yml]: Why run performance tests with a different python version (3.7 vs 3.6)?

File: .travis.yml (hunk 2 above)
Comment (vtomole) [reply .travis.yml]: Left it like this if we want to benchmark conrib programs (Do we want to?)

File: .travis.yml (hunk 3 above)
Comment (vtomole) [reply .travis.yml]: Wanted to have a variety of interpreters in travis cause users do; so it increases the chances of us catching version dependent problems. Now that you mention it, different interpreters can have different performances! I will make it consistent as you suggest.

File: .travis.yml (hunk 2 above)
Comment (viathor) [reply .travis.yml]: Our CI setup skips testing contrib for correctness (skipped above), so we should probably also skip testing contrib for performance.

File: .travis.yml (hunk 3 above)
Comment (viathor) [reply .travis.yml]: I see. Your argument about covering more interpreter versions makes sense to me. Either is OK.

**Comment by vtomole** [comment]
I tagged you for `pytest-benchmark` feedback based on the compatibility issues you've been experiencing.

File: .travis.yml (hunk 2 above)
Comment (Strilanc) [reply .travis.yml]: I disagree that we should skip contrib perf tests. The without contrib check is to ensure that we don't accidentally depend on contrib dependencies in the main codebase. We only need one check of this style.

**Comment by Strilanc** [comment]
Travis doesn't seem to be running the checks... weird.

I wonder if we tripped over a security feature, where they want to avoid people using it to mine bitcoins?

**Comment by Strilanc** [comment]
Ah, nope, the yaml doesn't parse:

image

File: .travis.yml [hunk 4]
```diff
@@ -46,4 +46,13 @@ matrix:
     install:
       - pip install -r requirements.txt
       - pip install -r dev_tools/conf/pip-list-dev-tools.txt
-    script: check/pytest --ignore=cirq/contrib
+    script: check/pytest --ignore=cirq/contrib --benchmark-skip
+
+
+    - os: linux
```
Comment (Strilanc) [inline .travis.yml]: This line is indented two spaces too far, and preventing the file from parsing.

Comment (vtomole) [reply .travis.yml]: Done.

**Pull request merged**